src/lib/datatypes/sol-vector.c

```diff
@@ -139,6 +139,34 @@ sol_vector_del(struct sol_vector *v, uint16_t i)
     return 0;
 }
 
+SOL_API int
+sol_vector_del_range(struct sol_vector *v, uint16_t start, uint16_t len)
+{
+    size_t tail_len;
+
```

Comment:
no sol-vector.c does that, so for consistency no.  However we should flag them as `SOL_ATTR_NON_NULL()` and add the checks so we're safe and we hint address sanitizer and optimizations can remove the checks. Can be done in a different commit.